### ISSUE TITLE
Prevent spreading of an array into an object

### ISSUE THREAD

**Issue by James-E-Adams** [palantir/tslint#4339]
### Rule Suggestion

**Is your rule for a general problem or is it specific to your development style?**

I think it's a general problem. I have buy in from one other person on twitter, so surely it's ubiquitous, right? https://twitter.com/jamesadams0/status/1069845022722277376

**What does your suggested rule do?**

Warns when you attempt to spread an array into an object.

**List several examples where your rule could be used**

```js
const a = [1,2,3]
const b = {...a}
````

**Additional context**

Playground: http://www.typescriptlang.org/play/#src=const%20a%20%3D%20%5B1%2C%202%2C%203%5D%0A%0Aconst%20b%20%3D%20%7B%20...a%20%7D%0A

**Comment by JoshuaKGoldberg** [comment]
😄 that didn't take long! Yes, this seems reasonable as a TSLint rule. The above code results in `{ '0': 1, '1': 2, '2': 3 }`, which could be surprising. Let's see if anybody else has suggestions or opposing thoughts.

**Comment by James-E-Adams** [comment]
Yeah it's perfectly valid javascript. But I think people do it 99% of the time without really wanting an object mapping indices to values.
 
😄Just a tab away!

**Comment by JoshuaKGoldberg** [comment]
Note: per #4534, this issue will be closed in less than a month if no PR is sent to add the rule. If you _really_ need the rule, custom rules are always an option and can be maintained outside this repo!

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is being deprecated** and no longer accepting pull requests for new rules. See #4534. 😱

If you'd like to see this rule implemented, you have two choices:
* **Recommended:** Use [typescript-eslint](https://typescript-eslint.io) to lint your TypeScript code with ESLint ✅
* **Legacy:** Write it as a [custom TSLint rule](https://palantir.github.io/tslint/develop/custom-rules/) 🤷‍♂️

👋 It was a pleasure open sourcing with you!

_If you believe this message was posted here in error, please comment so we can re-open the issue!_
